### ISSUE TITLE
OpenJ9 reproducibly fails at running an example working with Hotspot

### ISSUE THREAD

**Issue by DanySK** [eclipse/openj9#7207]
Java -version output
--------------------

The following versions are documented to be affected:

* On Linux
```
openjdk version "11" 2018-09-25
OpenJDK Runtime Environment AdoptOpenJDK (build 11+28)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.10.0, JRE 11 Linux amd64-64-Bit Compressed References 20181003_44 (JIT enabled, AOT enabled)
OpenJ9   - e44c4716
OMR      - 32df9563
JCL      - eeda2c2d02 based on jdk-11+28)
```
```
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment AdoptOpenJDK (build 12+33)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.13.0, JRE 12 Linux amd64-64-Bit Compressed References 20190320_41 (JIT enabled, AOT enabled)
OpenJ9   - caeb51f87
OMR      - 33a33ff2
JCL      - 419f571019 based on jdk-12+33)
```

* On Windows 10
```
openjdk version "11" 2018-09-25
OpenJDK Runtime Environment AdoptOpenJDK (build 11+28)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.10.0, JRE 11 Windows Server 2016 amd64-64-Bit Compressed References 20181003_41 (JIT enabled, AOT enabled)
OpenJ9   - e44c4716
OMR      - 32df9563
JCL      - eeda2c2d02 based on jdk-11+28)
```
```
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment AdoptOpenJDK (build 12+33)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.13.0, JRE 12 Windows Server 2016 amd64-64-Bit Compressed References 20190320_32 (JIT enabled, AOT enabled)
OpenJ9   - caeb51f87
OMR      - 33a33ff2
JCL      - 419f571019 based on jdk-12+33)
```

* On Mac OS X
```
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment AdoptOpenJDK (build 12+33)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.13.0, JRE 12 Mac OS X amd64-64-Bit Compressed References 20190320_32 (JIT enabled, AOT enabled)
OpenJ9   - caeb51f87
OMR      - 33a33ff2
JCL      - 419f571019 based on )
```

Strangely, the following is *not* affected under Mac OS X:
```
openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.1, JRE 11 Mac OS X amd64-64-Bit Compressed References 20190717_289 (JIT enabled, AOT enabled)
OpenJ9   - 0f66c6431
OMR      - ec782f26
JCL      - fa49279450 based on jdk-11.0.4+11)
```

Summary of problem
------------------

I got a test suite for a large project. Builds are run on a continuous integration platform.
At some point, builds began to selectively fail on OpenJ9:
https://travis-ci.org/AlchemistSimulator/Alchemist/builds/588970395

Issue stacktrace:
```
Unhandled exception
Type=Segmentation error vmState=0x00000000
J9Generic_Signal_Number=00000004 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
Handler1=00007F24419BFEB0 Handler2=00007F24412BC4B0 InaccessibleAddress=0000000099669A58
RDI=0000000001557EE8 RSI=00007F23A80712B4 RAX=00007F2422365BC0 RBX=00007F23A80712B3
RCX=00000000018C5000 RDX=0000000000000158 R8=00000000009B5F00 R9=00007F243CD4AE98
R10=00007F244054FCA0 R11=00000000EB350A70 R12=0000000099669900 R13=0000000001338200
R14=00007F2422365ED0 R15=0000000000030000
RIP=00007F244193C8D6 GS=0000 FS=0000 RSP=00007F2422365830
EFlags=0000000000010246 CS=0033 RBP=000000000138DB48 ERR=0000000000000004
TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=0000000099669A58
xmm0 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm1 43e0000000000000 (f: 0,000000, d: 9,223372e+18)
xmm2 3fc7565060000000 (f: 1610612736,000000, d: 1,823216e-01)
xmm3 43e0000000000000 (f: 0,000000, d: 9,223372e+18)
xmm4 40305de2ecc0a930 (f: 3972049152,000000, d: 1,636674e+01)
xmm5 3fd551f09652c188 (f: 2522006016,000000, d: 3,331262e-01)
xmm6 3fc999b324f10111 (f: 619774208,000000, d: 2,000030e-01)
xmm7 bff0000000000000 (f: 0,000000, d: -1,000000e+00)
xmm8 3ff336732d4de41f (f: 760079360,000000, d: 1,200793e+00)
xmm9 c010000000000000 (f: 0,000000, d: -4,000000e+00)
xmm10 3fdbb163511e49f0 (f: 1360939520,000000, d: 4,327019e-01)
xmm11 c018000000000000 (f: 0,000000, d: -6,000000e+00)
xmm12 c004c939771a7159 (f: 1998221696,000000, d: -2,598254e+00)
xmm13 3ef53ec8ec781fec (f: 3967295488,000000, d: 2,026105e-05)
xmm14 bb9a05c000000000 (f: 0,000000, d: -1,377618e-21)
xmm15 3fd9d80000000000 (f: 0,000000, d: 4,038086e-01)
Module=/home/danysk/.jabba/jdk/adopt-openj9@1.12.33-0/lib/compressedrefs/libj9vm29.so
Module_base_address=00007F2441921000
Target=2_90_20190320_41 (Linux 4.19.72-1-lts)
CPU=amd64 (12 logical CPUs) (0x7d08dc000 RAM)
```

The same tests run with OpenJDK

Diagnostic files
----------------

[javacore.20190924.173822.4050658.0002.txt](https://github.com/eclipse/openj9/files/3648083/javacore.20190924.173822.4050658.0002.txt)

[jitdump.20190924.173822.4050658.0004.tar.gz](https://github.com/eclipse/openj9/files/3648096/jitdump.20190924.173822.4050658.0004.tar.gz)

[Snap.20190924.173822.4050658.0003.trc.tar.gz](https://github.com/eclipse/openj9/files/3648092/Snap.20190924.173822.4050658.0003.trc.tar.gz)

Steps to reproduce
----------------

```bash
git clone git@github.com:AlchemistSimulator/Alchemist.git
git checkout c159976d20db55c0de21e6800b2789046160ed3f
cd alchemist
./gradlew alchemist-incarnation-biochemistry:test
```

**Comment by fjeremic** [comment]
From the javacore:

```
1XHEXCPMODULE  Module: /home/danysk/.jabba/jdk/adopt-openj9@1.12.33-0/lib/compressedrefs/libj9vm29.so
1XHEXCPMODULE  Module_base_address: 00007F2441921000
```

```
1XMCURTHDINFO  Current thread
3XMTHREADINFO      "Test worker" J9VMThread:0x0000000001338200, omrthread_t:0x00007F23B002EB18, java/lang/Thread:0x00000000E09C0AD8, state:R, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x20, isDaemon:false)
3XMTHREADINFO1            (native thread ID:0x3DCF0A, native priority:0x5, native policy:UNKNOWN, vmstate:R, vm thread flags:0x00000020)
3XMTHREADINFO2            (native stack address range from:0x00007F242232A000, to:0x00007F242236A000, size:0x40000)
3XMCPUTIME               CPU usage total: 4.977865493 secs, current category="Application"
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=8617688 (0x837ED8)
1INTERNAL                    Unable to obtain lock context information
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at (Missing Method)
3XMTHREADINFO3           Native callstack:
4XENATIVESTACK               (0x00007F24412D8452 [libj9prt29.so+0x3a452])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F24412D84CE [libj9prt29.so+0x3a4ce])
4XENATIVESTACK               (0x00007F24412D85C4 [libj9prt29.so+0x3a5c4])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F24412D832B [libj9prt29.so+0x3a32b])
4XENATIVESTACK               (0x00007F24412D49B2 [libj9prt29.so+0x369b2])
4XENATIVESTACK               (0x00007F24412D5754 [libj9prt29.so+0x37754])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F2440C97D66 [libj9dmp29.so+0x19d66])
4XENATIVESTACK               (0x00007F2440C97EFD [libj9dmp29.so+0x19efd])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F2440C9464D [libj9dmp29.so+0x1664d])
4XENATIVESTACK               (0x00007F2440C9016D [libj9dmp29.so+0x1216d])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F2440C90F8E [libj9dmp29.so+0x12f8e])
4XENATIVESTACK               (0x00007F2440C9A2AC [libj9dmp29.so+0x1c2ac])
4XENATIVESTACK               (0x00007F2440C82C3D [libj9dmp29.so+0x4c3d])
4XENATIVESTACK               (0x00007F2440C82255 [libj9dmp29.so+0x4255])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F2440C8585B [libj9dmp29.so+0x785b])
4XENATIVESTACK               (0x00007F2440C859DC [libj9dmp29.so+0x79dc])
4XENATIVESTACK               (0x00007F2440C9BECB [libj9dmp29.so+0x1decb])
4XENATIVESTACK               (0x00007F24419BFAA2 [libj9vm29.so+0x9eaa2])
4XENATIVESTACK               (0x00007F24412BD3C8 [libj9prt29.so+0x1f3c8])
4XENATIVESTACK               (0x00007F24419BFC96 [libj9vm29.so+0x9ec96])
4XENATIVESTACK               (0x00007F24412BC64F [libj9prt29.so+0x1e64f])
4XENATIVESTACK               (0x00007F2442E58D00 [libpthread.so.0+0x13d00])
4XENATIVESTACK               (0x00007F244193C8D6 [libj9vm29.so+0x1b8d6])
4XENATIVESTACK               (0x00007F2441939E10 [libj9vm29.so+0x18e10])
4XENATIVESTACK               (0x00007F24419FA8D2 [libj9vm29.so+0xd98d2])
```

**Comment by DanySK** [comment]
@fjeremic apologies, but it is not immediately clear to me what information the javacore snippet you posted is conveying -- if your post is directed to me.

**Comment by fjeremic** [comment]
> @fjeremic apologies, but it is not immediately clear to me what information the javacore snippet you posted is conveying -- if your post is directed to me.

Apologies @DanySK. I extracted some extra information from the javacore which says the crash occurred in libj9vm29.so and posted the native backtrace which seems to indicate a crash in the libj9prt29.so library. I posted this as justification for tagging this issue with the `comp:vm` label so that developers in this area are aware of this issue and can help investigate.

Given the crash appears to have happened in a native library what would help from your side is if you could load up the system core file against the JVM you used to run in gdb and extract the backtrace. I see the system core file is also not attached to the issue. We'll need this to debug the problem. Are you able to upload it somewhere and post the link here in the issue?

**Comment by DanySK** [comment]
@fjeremic thanks for the clarification.

I need some instructions on how to use gdb for extracting the backtrace. I'm not entirely sure if you refer to the libj9vm29.so file, in case, it's attached.
However, note that I'm just installing the version from the AdoptOpenJDK repository via Jabba.

[libj9vm29.tar.gz](https://github.com/eclipse/openj9/files/3648481/libj9vm29.tar.gz)

**Comment by pshipton** [comment]
@DanySK the initial description seems to indicate the problem occurs in the older OpenJ9 releases 0.10 and 0.13, but in resolved in the most recent release 0.15.1. If this holds true across all platforms, the problem is already resolved and the solution is to start using the latest release.

**Comment by DanySK** [comment]
Interesting... I was using Jabba for installing the "latest", and I did not realize they are actually using a very confusing versioning system, see shyiko/jabba#481.

I could force the version manually.

**Comment by DanySK** [comment]
@pshipton you are right. I circumvented shyiko/jabba#481 and the bug, whatever it was, is already fixed in OpenJ9 0.15.1.
https://travis-ci.org/AlchemistSimulator/Alchemist/builds/589118590

Thanks for your time
